File: packages/js-dpp/lib/errors/DataContractMaxDepthExceedError.js
const ConsensusError = require('./ConsensusError');

class DataContractMaxDepthExceedError extends ConsensusError {
  constructor() {
    super(`JSON Schema depth is greater than ${DataContractMaxDepthExceedError.MAX_DEPTH}`);
  }
}

DataContractMaxDepthExceedError.MAX_DEPTH = 500;

module.exports = DataContractMaxDepthExceedError;

File: packages/rs-dpp/src/errors/consensus/basic/data_contract_max_depth_exceed_error.rs
use crate::consensus::basic::BasicError;
use thiserror::Error;

#[derive(Error, Debug, Clone, PartialEq, Eq)]
#[error("JSON Schema depth is greater than {0}")]
pub struct DataContractMaxDepthExceedError {
    max_depth: usize
}

impl DataContractMaxDepthExceedError {
    pub fn new(max_depth: usize) -> Self {
        Self {
            max_depth,
        }
    }

    pub fn max_depth(&self) -> usize {
        self.max_depth
    }
}

impl From<DataContractMaxDepthExceedError> for BasicError {
    fn from(err: DataContractMaxDepthExceedError) -> Self {
        Self::DataContractMaxDepthExceedError(err)
    }
}
